### DISCUSSION TITLE
Adding additional XML namespaces

### DISCUSSION THREAD

**Discussion by nemec** [cduck/drawSvg#60] [Q&A] [answered]
Is it possible to add additional XML namespace definitions prior to writing out the shapes to XML? It's easy enough to add shapes with namespaces, e.g.

    draw.Group(id='layer-graph-entities', inkscape__groupmode='layer', inkscape__label='Graph Entities')

But the namespace definition is missing from the final output, which breaks things like my default image viewer.

I see it's being hardcoded here, but maybe there's an opportunity to override elsewhere?

https://github.com/cduck/drawSvg/blob/master/drawSvg/drawing.py#L124

I'm able to work around it by rewriting the XML with `lxml` to add the namespace, but it's really horrible code:

```python
import xml.etree.ElementTree as ET
from lxml import etree

ET.register_namespace('inkscape', 'http://www.inkscape.org/namespaces/inkscape')
ET.register_namespace('', "http://www.w3.org/2000/svg")
parser = etree.XMLParser(recover=True)
tree = ET.parse('floorplan.svg', parser)
root = tree.getroot()
root.attrib['{http://www.inkscape.org/namespaces/inkscape}version'] = 'none'

tree.write('floorplan.svg')
```

**Comment by cduck** [comment, marked as answer]
Custom namespaces are not currently supported but could be in the future.
